File: _posts/Programmers-main/프로그래머스/unrated/숫자 변환하기/숫자 변환하기.py
def solution(x, y, n):
    answer = 0
    listt = [y]
    while x not in listt:
        temp = []
        for i in listt:
            if i % 3 == 0:
                temp.append(i / 3)
            if i % 2 == 0:
                temp.append(i / 2)
            if i-n >= x:
                temp.append(i-n)
        answer += 1
        listt = temp
        if len(temp) == 0:
            answer = -1
            break
    return answer

File: _posts/Programmers_cpp-main/프로그래머스/unrated/숫자 변환하기/숫자 변환하기.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(int x, int y, int n) {
    int answer = 0;
    set<int> list = {x};
    while (true)
    {
        set<int> temp = {};
        for (auto ele : list)
        {
            if (ele == y) return answer;
            if ((ele * 2) <= y) temp.insert(ele * 2);
            if ((ele * 3) <= y) temp.insert(ele * 3);
            if ((ele + n) <= y) temp.insert(ele + n);
        }
        if (temp.size() == 0) return answer = -1;
        list = temp;
        answer += 1;
    }
}